src/services/codeFixProvider.ts

```diff
@@ -127,6 +127,7 @@ function getDiagnostics({ program, sourceFile, cancellationToken }: CodeFixConte
     return [
         ...program.getSemanticDiagnostics(sourceFile, cancellationToken),
         ...program.getSyntacticDiagnostics(sourceFile, cancellationToken),
+        ...program.getDeclarationDiagnostics(sourceFile, cancellationToken),

```

Comment:
Curious how much this slows down quick fixes; hopefully not too much.

Reply:
Could we conditionally call this based on the compilerOptions isolatedDeclarations setting?

Reply:
I was going to suggest the same - we should probably only do this depending on settings.

Reply:
That being said, I don't think it actually matters; I'm guessing this will be cached given you'd already have declarations enabled and see the diagnostics in the editor.

Reply:
applied the suggestion

Reply:
I actually don't think you need to do that; it looks like the results are cached anyway. Though reading other places that call this function, it does make me wonder if we should just be merging the idea of a declaration error into semantic errors...

Reply:
Ah, okay. I'll wait for more feedback on your idea - or do you think it's rather something that's orthogonal to this PR?

Reply:
Yeah, I'd say it's orthogonal; note how the LS does this: https://github.com/microsoft/TypeScript/blob/21b5c9624c0765e6baea1f21a26531aac4824a41/src/services/services.ts#L2038

The relevant thing is https://github.com/microsoft/TypeScript/blob/21b5c9624c0765e6baea1f21a26531aac4824a41/src/compiler/program.ts#L2884 and https://github.com/microsoft/TypeScript/blob/21b5c9624c0765e6baea1f21a26531aac4824a41/src/compiler/program.ts#L3281 given this implies that it's already cached. However, it looks like everyone conditionally calls this based on whether or not declaration emit is enabled, which seems to imply that leaving it conditional may be right (or conditionalize it on --declararions). But clearly we haven't had any past declaration emit quick fixes.

So... I think the state of the PR (checking ahead of time) is actually correct. But you can see why I'm thinking that conditionalizing everywhere is probably no longer super clear.